src/alejandra/src/rules/inherit.rs

```diff
@@ -53,7 +53,15 @@ pub(crate) fn rule(
                 steps.push_back(crate::builder::Step::Whitespace);
                 steps.push_back(crate::builder::Step::Comment(text));
                 steps.push_back(crate::builder::Step::NewLine);
-                steps.push_back(crate::builder::Step::Pad);
+                // Only add padding if there are no `trivialities` (that is, there's no extra
+                // `Newlines(_)` to be added) or if the first one is a comment (that is, it'll need
+                // to be indented to match the content).
+                if matches!(
+                    child.trivialities.front(),
+                    None | Some(crate::children2::Trivia::Comment(_))
+                ) {
```

Comment:
I'm not super thrilled with the complexity in this function. I feel like we could restructure this to be significantly cleaner, though I'm not sure how. Maybe we could more liberally generate `NewLine`s and `Pad`s and then have another pass through the `Step`s at the end to remove trailing whitespace, consecutive blank lines, blank lines before semicolons, that sorta thing?

Reply:
yes, the current code is certainly not very readable

I'm not sure if upgrading rnix-parser would help with this, the latest version apparently includes some improvements to the API

as far as I can tell, the important checks for me are:
- https://github.com/kamadorueda/alejandra/blob/main/buildkite.yaml#L35
- https://github.com/kamadorueda/alejandra/blob/main/buildkite.yaml#L60

since that guarantee that the formatting is stable across Alejandra versions (or changed for a good reason) and that formatted code is valid nix code. If those 2 pass, I'm fine with changing the code to make it more readable or making the formatted code nicer